var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -163,28 +168,38 @@ def determine_b2_options(self, spec, options):
             threadingOpts.append('single')
         if not threadingOpts:
             raise RuntimeError("At least one of {singlethreaded, multithreaded} must be enabled")
-        
+
         options.extend([
             'toolset=%s' % self.determine_toolset(spec),
             'link=%s' % ','.join(linkTypes),
             '--layout=tagged'])
-        
+
         return threadingOpts
 
     def install(self, spec, prefix):
+        # On Darwin, Boost expects the Darwin libtool. However, one of the
+        # dependencies may have pulled in Spack's GNU libtool, and these two are
+        # not compatible. We thus create a symlink to Darwin's libtool and add
+        # it at the beginning of PATH.
+        if sys.platform == 'darwin':
```

Comment:
i guess the same issue may appear in other places? Would that be better to force OS-X users somehow do something like
```
packages:
  all:
    compiler: [clang, gcc]
  libtool:
    paths:
      libtool@2.4.6%clang@7.0.2-apple: /usr/ # libtool is in /usr/bin/
    buildable: False # never build libtool
```
in `packages.yaml`? Maybe not only libtool, but some reasonable defaults for things which ship with OS-X XCode, i.e. `zlib`, etc?

I think it is also related to this issue https://github.com/LLNL/spack/issues/541